Source/Core/Core/PowerPC/Tables/TableGen.cpp

```diff
@@ -0,0 +1,476 @@
+  {
+    for (size_t i = end; i < lines.size() && !std::holds_alternative<StartTableMarker>(lines[i]);
+         i += 1)
+    {
+      if (!std::holds_alternative<Empty>(lines[i]))
+      {
+        std::cerr << "Error: subtable \"" << entry.table_name << "\" (" << entry.length
+                  << "-bit field) is longer than " << (1 << entry.length) << "\n";
+        std::exit(1);
+      }
+    }
+  }
+  entry.instructions <<= 64 + start - end;
+  entry.subtables <<= 64 + start - end;
+  const int offset = entry.subtable_offset;
+  for (auto& i : subtables)
+  {
+    // WARNING: may invalidate 'entry'
+    decoding_table.push_back(DecodingEntry(std::get<SubTable>(lines[i]).name));
+  }
+  for (size_t i = 0; i < subtables.size(); i += 1)
+  {
+    CreateTable(table, decoding_table, decoding_table[offset + i], lines);
+  }
+}
+
+static void DoOutput(const std::vector<Instruction>& table,
+                     const std::vector<DecodingEntry>& decoding_table, const OutputOptions& options,
+                     std::ostream& out)
+{
+  if ((options.flags & OPID_RANGES) || (options.flags & DECODING_TABLE) || !options.columns.empty())
+  {
+    out << "// Auto-generated by TableGen.cpp\n// clang-format off\n";
+  }
+  if (options.flags & OPID_RANGES)
+  {
+    for (auto& dec : decoding_table)
+    {
+      out << dec.table_name << " = " << dec.instruction_offset + 1 << ", // " << dec.description
+          << "\n";
+      out << dec.table_name << "_End = " << dec.instruction_offset + 1 + dec.instructions.count()
+          << ",\n";
+    }
+    out << "End = " << table.size() + 1 << '\n';
+  }
+  if (options.flags & DECODING_TABLE)
+  {
+    for (auto& dec : decoding_table)
+    {
+      out << "{0b" << dec.instructions << ", 0b" << dec.subtables << ", "
+          << dec.instruction_offset + 1 << ", " << dec.subtable_offset << ", " << dec.shift << ", "
+          << dec.length << "},\n";
+    }
+  }
+  if (!options.columns.empty())
+  {
+    for (auto& inst : table)
+    {
+      if (options.columns.size() > 1)
+      {
+        out << "{";
+      }
+      for (auto& col : options.columns)
+      {
+        if (inst.cells.size() <= col.column || inst.cells[col.column].empty())
+        {
+          if (col.default_value == "*")
+          {
+            out << col.prefix << inst.cells.at(0) << col.suffix << ", ";
+          }
+          else
+          {
+            out << col.default_value << ", ";
+          }
+        }
+        else if (inst.cells[col.column] == "*")
+        {
+          out << col.prefix << inst.cells[0] << col.suffix << ", ";
+        }
+        else
+        {
+          out << col.prefix << inst.cells[col.column] << col.suffix << ", ";
+        }
+      }
+      if (options.columns.size() > 1)
+      {
+        out << "},";
+      }
+      out << '\n';
+    }
+  }
+  if ((options.flags & OPID_RANGES) || (options.flags & DECODING_TABLE) || !options.columns.empty())
+  {
+    out << "// clang-format on\n";
```

Comment:
The committed files have this line indented by four spaces; if clang-format insists on that then the generator should include that.